### DISCUSSION TITLE
HIP-20: Standardized URI scheme to reference Hedera data

### DISCUSSION THREAD

**Discussion by sergmetelin** [hiero-ledger/hiero-improvement-proposals#110] [Edits]
This discussion is designated as the discussions-to target of [hip-20](https://github.com/hashgraph/hedera-improvement-proposal/blob/master/HIP/hip-20.md). Use this thread to further develop the HIP and move it through to its final lifecycle state.

**Comment by sergmetelin** [comment]
@raykhoaza @sam-at-luther would the implementation in https://hips.hedera.com/hip/hip-30 address the functionality described in hip-20?

**Reply by sam-at-luther** [reply]
It looks like it's accomplishing a similar goal. If it is expanded to include references to 1) HCS topics and 2) HFS files, then that specification would make this one redundant.

**Reply by sergmetelin** [reply]
@shemnon adding you to this thread to align on the expectations

**Reply by shemnon** [reply]
Should be fairly easy.  Create "assets" for topic and file.  `hedera:mainnet/topic:0.0.123-vfmkw` and `hedera:mainnet/file:0.0.123-vfmkw`

**Reply by shemnon** [reply]
Note however that the CAIP editors are hibernating and are not reviewing the hedera namespace CAIPs.

**Reply by sam-at-luther** [reply]
I'm fine with closing this proposal, if HIP-30 is updated to include `topic` and `file` as described here

**Reply by shemnon** [reply]
https://github.com/hashgraph/hedera-improvement-proposal/pull/216

**Comment by sam-at-luther** [comment]
HIP-10 has a [PR](https://github.com/hashgraph/hedera-improvement-proposal/pull/217) to reference HIP-30 CAIDs for metadata URIs. This HIP is subsumed by HIP-30 and IMO can be abandoned.
